java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtAudienceValidator.java

```diff
@@ -94,6 +93,15 @@ private ValidationResult validateAudienceOfXsuaaBrokerClone(Set<String> allowedA
 				audiences.add(audience);
 			}
 		}
+		// extract audience (app-id) from scopes
+		if (audiences.size() == 0) {
+			for (String scope : token.getClaimAsStringList(TokenClaims.XSUAA.SCOPES)) {
+				if (scope.contains(".")) {
+					String aud = scope.substring(0, scope.indexOf('.'));
```

Comment:
This looks a lot like the code in line 88 but without calling `trim`.
Is `trim` not needed here? If it is I would suggest to extract that into a helper that strips the namespace. This can be used here and on line 89 then.

Reply:
Done